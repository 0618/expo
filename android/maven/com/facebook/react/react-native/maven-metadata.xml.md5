654f545cad70b83df5e4c0a788c23123