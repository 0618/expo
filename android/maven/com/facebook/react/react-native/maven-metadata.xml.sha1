fe12994307817f9ba93746f2777b7258b695a8f1