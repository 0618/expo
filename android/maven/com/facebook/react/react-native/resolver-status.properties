#NOTE: This is an internal implementation file, its format can be changed without prior notice.
#Fri Dec 08 19:01:09 PST 2017
maven-metadata-remote.xml.lastUpdated=1512788469554
