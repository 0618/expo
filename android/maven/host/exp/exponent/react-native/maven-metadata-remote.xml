<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>host.exp.exponent</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>13.0.0</release>
    <versions>
      <version>0.36.0</version>
      <version>12.0.0</version>
      <version>13.0.0</version>
    </versions>
    <lastUpdated>20170106221733</lastUpdated>
  </versioning>
</metadata>
