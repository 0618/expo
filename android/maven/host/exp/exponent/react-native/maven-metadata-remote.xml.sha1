21958b89d38ce436c8632fec92cafa5cfd95932e