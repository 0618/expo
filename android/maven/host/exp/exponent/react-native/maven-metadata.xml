<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>host.exp.exponent</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>16.0.0</release>
    <versions>
      <version>16.0.0</version>
    </versions>
    <lastUpdated>20170417172233</lastUpdated>
  </versioning>
</metadata>
