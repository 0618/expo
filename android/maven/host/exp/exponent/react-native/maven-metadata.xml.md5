8406df1d2c62c7a02fb4e0becf9092d8