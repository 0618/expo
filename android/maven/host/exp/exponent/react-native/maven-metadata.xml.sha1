87bd7ba66f591a4e3b177325e9e548698644d4dc