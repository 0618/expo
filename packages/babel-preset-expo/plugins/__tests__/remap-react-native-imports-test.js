const plugin = require('../remap-react-native-imports');
const pluginTester = require('babel-plugin-tester').default;

const tests = [
  // import react-native
  {
    title: 'import from "native-native"',
    code: `import ReactNative from 'react-native';
import { View, Pressable, findNodeHandle } from 'react-native';
import { Invalid, View as MyView } from 'react-native';
import * as ReactNativeModules from 'react-native';`,
    snapshot: true,
  },
  {
    title: 'import from "native-native"',
    code: `import ReactNative from 'react-native';
import { View, Pressable, findNodeHandle } from 'react-native';
import { Invalid, View as MyView } from 'react-native';
import * as ReactNativeModules from 'react-native';`,
    snapshot: true,
    pluginOptions: { commonjs: true },
  },
  {
    title: 'export from "react-native"',
    code: `export { View, findNodeHandle } from 'react-native';
export { StyleSheet, Text } from 'react-native';`,
    snapshot: true,
  },
  {
    title: 'require "react-native"',
    code: `const ReactNative = require('react-native');
const { View, Pressable, findNodeHandle } = require('react-native');
const { StyleSheet, TouchableOpacity } = require('react-native');`,
    snapshot: true,
  },
  {
    title: 'require "react-native"',
    code: `const ReactNative = require('react-native');
const { View, Pressable, findNodeHandle } = require('react-native');
const { StyleSheet, TouchableOpacity } = require('react-native');`,
    snapshot: true,
    pluginOptions: { commonjs: true },
  },
];

pluginTester({
  babelOptions: {
    generatorOpts: {
      jsescOption: {
        quotes: 'single',
      },
    },
  },
  plugin,
  pluginName: 'Rewrite react-native to internal imports',
  tests,
});
