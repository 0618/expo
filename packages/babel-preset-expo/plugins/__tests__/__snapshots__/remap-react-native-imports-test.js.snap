// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewrite react-native to internal imports export from "react-native": export from "react-native" 1`] = `

export { View, findNodeHandle } from 'react-native';
export { StyleSheet, Text } from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

export { default as View } from 'react-native/Libraries/Components/View/View';
export { findNodeHandle } from 'react-native/Libraries/Renderer/shims/ReactNative';
export { default as StyleSheet } from 'react-native/Libraries/StyleSheet/StyleSheet';
export { default as Text } from 'react-native/Libraries/Text/Text';


`;

exports[`Rewrite react-native to internal imports import from "native-native": import from "native-native" 1`] = `

import ReactNative from 'react-native';
import { View, Pressable, findNodeHandle } from 'react-native';
import { Invalid, View as MyView } from 'react-native';
import * as ReactNativeModules from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import ReactNative from 'react-native/index';
import View from 'react-native/Libraries/Components/View/View';
import Pressable from 'react-native/Libraries/Components/Pressable/Pressable';
import { findNodeHandle } from 'react-native/Libraries/Renderer/shims/ReactNative';
import { Invalid } from 'react-native/index';
import MyView from 'react-native/Libraries/Components/View/View';
import * as ReactNativeModules from 'react-native/index';


`;

exports[`Rewrite react-native to internal imports import from "native-native": import from "native-native" 2`] = `

import ReactNative from 'react-native';
import { View, Pressable, findNodeHandle } from 'react-native';
import { Invalid, View as MyView } from 'react-native';
import * as ReactNativeModules from 'react-native';

      ↓ ↓ ↓ ↓ ↓ ↓

import ReactNative from 'react-native/index';
import View from 'react-native/Libraries/Components/View/View';
import Pressable from 'react-native/Libraries/Components/Pressable/Pressable';
import { findNodeHandle } from 'react-native/Libraries/Renderer/shims/ReactNative';
import { Invalid } from 'react-native/index';
import MyView from 'react-native/Libraries/Components/View/View';
import * as ReactNativeModules from 'react-native/index';


`;

exports[`Rewrite react-native to internal imports require "react-native": require "react-native" 1`] = `

const ReactNative = require('react-native');
const { View, Pressable, findNodeHandle } = require('react-native');
const { StyleSheet, TouchableOpacity } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

const ReactNative = require('react-native/index');

const View = require('react-native/Libraries/Components/View/View');

const Pressable = require('react-native/Libraries/Components/Pressable/Pressable').default;

const findNodeHandle = require('react-native/Libraries/Renderer/shims/ReactNative').findNodeHandle;

const StyleSheet = require('react-native/Libraries/StyleSheet/StyleSheet');

const TouchableOpacity = require('react-native/Libraries/Components/Touchable/TouchableOpacity');


`;

exports[`Rewrite react-native to internal imports require "react-native": require "react-native" 2`] = `

const ReactNative = require('react-native');
const { View, Pressable, findNodeHandle } = require('react-native');
const { StyleSheet, TouchableOpacity } = require('react-native');

      ↓ ↓ ↓ ↓ ↓ ↓

const ReactNative = require('react-native/index');

const View = require('react-native/Libraries/Components/View/View');

const Pressable = require('react-native/Libraries/Components/Pressable/Pressable').default;

const findNodeHandle = require('react-native/Libraries/Renderer/shims/ReactNative').findNodeHandle;

const StyleSheet = require('react-native/Libraries/StyleSheet/StyleSheet');

const TouchableOpacity = require('react-native/Libraries/Components/Touchable/TouchableOpacity');


`;
